### ISSUE TITLE
String index out of bounds using JvmOverloads and config file

### ISSUE THREAD

**Issue by FWDekker** [detekt/detekt#2166]
With the detekt formatting plugin enabled, an exception is thrown when running `gradlew detekt` if a file uses the `@JvmOverloads` operator on a constructor _and_ a config file has been configured for detekt.

## Expected Behavior
No exception is thrown because the code and configuration are all valid.

## Observed Behavior
<details>
  <summary>The following exception is thrown. (Click here.)</summary>
  <p>

```java
.Analyzing '/home/felix/Projects/detekt-test/src/main/kotlin/com/fwdekker/detekttest/DetektTest.kt' led to an exception.
The original exception message was: String index out of range: -1
Running detekt '1.2.1' on Java '11.0.4+11-post-Ubuntu-1ubuntu218.04.3' on OS 'Linux'.
If the exception message does not help, please feel free to create an issue on our GitHub page.
java.base/java.lang.String.substring(String.java:1837)
com.pinterest.ktlint.ruleset.experimental.AnnotationRule.visit(AnnotationRule.kt:97)
io.gitlab.arturbosch.detekt.formatting.FormattingRule.apply(FormattingRule.kt:61)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule$visit$2$$special$$inlined$forEach$lambda$1.invoke(KtLintMultiRule.kt:88)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule$visit$2$$special$$inlined$forEach$lambda$1.invoke(KtLintMultiRule.kt:46)
io.gitlab.arturbosch.detekt.api.MultiRule.runIfActive(MultiRule.kt:33)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule$visit$2.invoke(KtLintMultiRule.kt:88)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule$visit$2.invoke(KtLintMultiRule.kt:46)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule.visitTokens(KtLintMultiRule.kt:101)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule.visitTokens(KtLintMultiRule.kt:103)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule.visitTokens(KtLintMultiRule.kt:103)
io.gitlab.arturbosch.detekt.formatting.KtLintMultiRule.visit(KtLintMultiRule.kt:86)
io.gitlab.arturbosch.detekt.api.BaseRule.visitFile(BaseRule.kt:43)
io.gitlab.arturbosch.detekt.api.RuleSet.accept(RuleSet.kt:35)
io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:76)
io.gitlab.arturbosch.detekt.core.Detektor.runSync(Detektor.kt:44)
io.gitlab.arturbosch.detekt.core.Detektor.run(Detektor.kt:28)
io.gitlab.arturbosch.detekt.core.DetektFacade.run(DetektFacade.kt:41)
io.gitlab.arturbosch.detekt.cli.runners.Runner.execute(Runner.kt:29)
io.gitlab.arturbosch.detekt.cli.Main.main(Main.kt:16)
```

  </p>
</details>

After this exception, detekt concludes that no errors have been detected, even though the source file in question contains several.

## Steps to Reproduce
1. [Download this archive of a minimal working example.](https://github.com/arturbosch/detekt/files/3927820/detekt-issue-mwe.zip)
2. Run `gradlew detekt`

Observe that each of the following actions stop the issue (but are not valid solutions for my use case):
* Remove the `@JvmOverloads` from `DetektTest.kt`.
* Remove `config = files("detekt.yml")` from `build.gradle`.
* Remove the dependency on the detekt-formatting plugin from `build.gradle`.

## Context
The exception comes up whenever I let detekt check my project. As a result, there may be style errors

## Your Environment
* Version of detekt used: 1.2.1 or 1.2.2
* Version of Gradle used (if applicable): 6.0.1
* Version of Java used: either 1.8.0_222 or 11.0.4
* Operating System and version: Ubuntu 18.04.3 LTS
* Link to your project (if it's a public repository): [intellij-randomness](https://github.com/FWDekker/intellij-randomness)

**Comment by schalkms** [comment]
Does this exception also occur, if you use the 'plain' `ktlint` library?

**Comment by FWDekker** [comment]
The exception does not occur if I use plain ktlint.

I tried to verify that as follows: In the minimal working example above, I removed the detekt-formatting plugin and added v9.1.1 of the [ktlint plugin](https://plugins.gradle.org/plugin/org.jlleitschuh.gradle.ktlint). Running `./gradlew ktlintCheck` was successful: It correctly detected a style error in my `DetektTest.kt` file. Running `./gradlew ktlintFormat` correctly fixes the style error.

**Comment by biazmoreira** [comment]
Hey, @FWDekker and @arturbosch!

I saw this issue is listed in the _Next_ in the _Bugs and False Positives_ board and tried to tackle it. 

The [link to the minimal working example is not working](https://github.com/arturbosch/detekt/files/3927820/detekt-issue-mwe.zip) anymore. I tried to reproduce the issue in a project of my own and also tried using the project [you linked](intellij-randomness), but I was not able to. Since this issue was created a while ago and [FormattingRule](https://github.com/detekt/detekt/blob/master/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt) has gone through a lot of changes since then, I guess this issue has been unintentionally solved. 

Since running "plain" `ktlint` doesn't give you any exceptions, I'm also guessing this is not a problem with the [AnnotationRule](https://github.com/pinterest/ktlint/blob/master/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/AnnotationRule.kt).

My take here is that the issue can be closed and be considered solved. :)

**Comment by arturbosch** [comment]
Thanks for the further investigation! I will close then for now.

**Comment by FWDekker** [comment]
Thanks for the investigation, @biazmoreira! I think the repo was renamed from arturbosch/detekt to detekt/detekt, so the link to the mwe has also been changed. I've updated the issue description so that the link is correct again.

Running the mwe shows the original exception. Updating the detekt version to 1.15.0 and changing `detekt.yml` to suppress `AbsentOrWrongFileLicense` and `UndocumentedPublicClass` are all that were needed to solve the issue, so I think it's safe to say that the issue has been resolved.

Thanks!
